src/oss-find-squats-lib/MutateExtension.cs

```diff
@@ -104,32 +104,54 @@ public static class MutateExtension
 
             Dictionary<string, IList<Mutation>> generatedMutations = new();
 
-            // Check to see if it is a scoped npm package to generate candidates for.
-            bool isScoped = purl.HasNamespace();
-            string nameToMutate = isScoped ? purl.Namespace : purl.Name;
+            // Check to see if the package has a namespace to generate candidates for.
+            bool hasNamespace = purl.HasNamespace();
+            string nameToMutate = hasNamespace ? purl.Namespace : purl.Name;
 
             foreach (IMutator mutator in mutators)
             {
                 foreach (Mutation mutation in mutator.Generate(nameToMutate))
                 {
-                    // Construct the mutated name if the package was scoped.
-                    string? mutationNamespace = isScoped ? mutation.Mutated : null;
-                    string mutationName = isScoped ? purl.Name : mutation.Mutated;
-                    PackageURL mutatedPurl = new(purl.Type, mutationNamespace, mutationName, null, null, null);
-                    string mutatedPurlString = mutatedPurl.ToString();
+                    // Construct the mutated name if the package has a namespace.
+                    Mutation newMutation = CreateMutation(purl, hasNamespace, mutation);
                     
-                    if (!generatedMutations.ContainsKey(mutatedPurlString))
+                    if (!generatedMutations.ContainsKey(newMutation.Mutated))
                     {
-                        generatedMutations[mutatedPurlString] = new List<Mutation>();
+                        generatedMutations[newMutation.Mutated] = new List<Mutation>();
                     }
 
-                    generatedMutations[mutatedPurlString].Add(new Mutation(mutatedPurlString, purl.ToString(), mutation.Reason, mutation.Mutator));
+                    generatedMutations[newMutation.Mutated].Add(newMutation);
                 }
             }
 
+            if (!hasNamespace)
+            {
+                return generatedMutations;
+            }
+
+            PackageURL purlWithoutNamespace = 
```

Comment:
Why not make this its own mutate extension?

Reply:
I created a RemoveNamespaceMutator. It's kind of difficult to handle something like this because the design just mutates the namespace if there is one, so the mutators never get the full name (namespace+name). So I check after the if(!hasNamespace)